### ISSUE TITLE
Fix focus retention of tabs while switching between IDEs

### ISSUE THREAD

**Issue by ankitakinger** [appsmithorg/appsmith#37648]
When you switch between IDEs, the old IDE loses the focus retention of tabs. For example, you're on app editor and on page 2, with 6 queries/apis open (tabs). Now, you switch to home screen and open a package editor, and then back to the app editor, the opened page as well as its open tabs are no longer retained. 

This also happens when you simply move from editor to homepage and back to editor. Another scenario this happens is on page refresh. And yet another one, if you try to connect and authorize a Google spreadsheet as a datasource, once you are back on the app editor after authorisation is complete, the opened tabs are no longer retained (open bug [#31590](https://github.com/appsmithorg/appsmith/issues/31590))

This needs to be fixed by maybe using local storage or you can explore other solutions.

Also, handle the state that maintains which tabs are open so that when user switches between say data and editor tabs in the sidebar, the open entities should remain open. The focus retention across IDEs will be handled in another task.

**Comment by hetunandu** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/IDE-Pod-60507ad1d4b98d00150a2858/issues/appsmithorg/appsmith/37648?planning-poker) with Zenhub @alex-golovanov @ankitakinger
